src/App.tsx

```diff
@@ -1,24 +1,173 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { v4 as uuid } from 'uuid';
+import { deleteTodo, getTodos, postTodo } from './api/todos';
+import { Error } from './components/Error/Error';
+import { Filter } from './components/Filter/Filter';
+import { Item } from './components/Todo/Todo';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 6470;
 
 export const App: React.FC = () => {
+  const [titleTodo, setTitleTodo] = useState('');
+  const [listTodo, setListTodo] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [tempTodo, setTempTodo] = useState(false);
+  const [showFooter, setShowFooter] = useState(true);
+  const [length, setLength] = useState(0);
+
+  const getListTodo = async (filter?: boolean, type?: string) => {
+    try {
+      const result = await getTodos(USER_ID, filter);
+
+      if (type === 'active') {
+        setListTodo(result.filter((el:Todo) => !el.completed));
+      } else {
+        setListTodo(result);
+      }
+    } catch (e) {
+      setError('Oops, something were wrong, please try again later');
+    }
+  };
+
+  useEffect(() => {
+    getListTodo();
+  }, []);
+
+  useEffect(() => {
+    getTodos(USER_ID).then(res => {
+      setLength(res.length);
+      if (!res.length) {
+        setShowFooter(false);
+      } else {
+        setShowFooter(true);
+      }
+    });
```

Comment:
why do you need this second useEffect if you call getTodos in `getListTodo`

Reply:
because like this i count all elements in getListTodo i have only todo which i choose in filter, i make request on back where choose type filter

Reply:
It looks a little bit confused. You should have one `useEffect` hook for getting todos. If you have some dynamic elements like filter_type or something you can use function params or hook dependencies